src/Simplex/Chat.hs

```diff
@@ -192,10 +192,13 @@ startChatController user subConns enableExpireCIs = do
     runExpireCIs = do
       let interval = 1800 * 1000000 -- 30 minutes
       forever $ do
-        expire <- asks expireCIs
-        atomically $ readTVar expire >>= \b -> unless b retry
-        ttl <- withStore' (`getChatItemTTL` user)
-        forM_ ttl $ \t -> expireChatItems user t False
+        ( do
```

Comment:
make a function - e.g. runExpireCIs could be this, and the remaining code just used where it's called

Reply:
alternatively and may be better use handleError